# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaTrackGraph.h
# Commit: 8cb7b8231c98
# Full Hash: 8cb7b8231c980aa7830fad2ecc34cb128417d511
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2022-01-28 04:41:16
# Regressor Bug: 1742655
# File Overlap Count: 1
# Description:
#   Bug 1742655 - Open/Close audio input via NativeInputTrack r=pehrsons
#   
#   Move the audio-opening check logic and the creation of NativeInputTrack
#   in one place instead of seperating them in different functions. This
#   reduces the complexity of audio-opening logic in MediaTrackGraph and
# ==============================================================================

diff -r a1def130c105 -r 8cb7b8231c98 dom/media/MediaTrackGraph.h
--- a/dom/media/MediaTrackGraph.h	Thu Jan 27 22:29:34 2022 +0000
+++ b/dom/media/MediaTrackGraph.h	Thu Jan 27 22:29:35 2022 +0000
@@ -1030,10 +1030,8 @@
   // Idempotent
   void ForceShutDown();
 
-  virtual nsresult OpenAudioInput(CubebUtils::AudioDeviceID aID,
-                                  AudioDataListener* aListener) = 0;
-  virtual void CloseAudioInput(CubebUtils::AudioDeviceID aID,
-                               AudioDataListener* aListener) = 0;
+  virtual void OpenAudioInput(NativeInputTrack* aTrack) = 0;
+  virtual void CloseAudioInput(NativeInputTrack* aTrack) = 0;
 
   // Control API.
   /**